programs/mango-v4/src/instructions/openbook_v2_cancel_order_by_client_order_id.rs

```diff
@@ -0,0 +1,108 @@
+use anchor_lang::prelude::*;
+
+use openbook_v2::cpi::accounts::CancelOrder;
+
+use crate::error::*;
+use crate::logs::{emit_stack, OpenbookV2OpenOrdersBalanceLog};
+use crate::serum3_cpi::OpenOrdersAmounts;
+use crate::serum3_cpi::OpenOrdersSlim;
+use crate::state::*;
+
+use crate::accounts_ix::*;
+
+use openbook_v2::state::Side as OpenbookV2Side;
+
+// Very similar to openbook_v2_cancel_order except this uses a u64 and calls the
+// cpi for cancelling by client order id. Uses the same accounts.
+pub fn openbook_v2_cancel_order_by_client_order_id(
+    ctx: Context<OpenbookV2CancelOrder>,
+    side: OpenbookV2Side,
+    client_order_id: u64,
+) -> Result<()> {
+    // Check instruction gate. Guarded by the same as cancelling by exchange id.
+    let group = ctx.accounts.group.load()?;
+    require!(
+        group.is_ix_enabled(IxGate::OpenbookV2CancelOrder),
```

Comment:
For perp we have a dedicated ix_gate for cancel by client id, maybe add one here ?

Reply:
could if you want. dont do it in serum3 though.
would require duplicating the account_ix file because it would be a new constraint

Reply:
ok can probably keep like this, thx